pypeit/find_objects.py

```diff
@@ -0,0 +1,1330 @@
+        extractmask (`numpy.ndarray`_):
+            Extraction mask
+        slits (:class:`pypeit.slittrace.SlitTraceSet`):
+        sobjs_obj (:class:`pypeit.specobjs.SpecObjs`):
+            Only object finding but no extraction
+        sobjs (:class:`pypeit.specobjs.SpecObjs`):
+            Final extracted object list with trace corrections applied
+        spat_flexure_shift (float):
+        tilts (`numpy.ndarray`_):
+            WaveTilts images generated on-the-spot
+        waveimg (`numpy.ndarray`_):
+            WaveImage image generated on-the-spot
+        slitshift (`numpy.ndarray`_):
+            Global spectral flexure correction for each slit (in pixels)
+        vel_corr (float):
+            Relativistic reference frame velocity correction (e.g. heliocentyric/barycentric/topocentric)
+
+    """
+
+    __metaclass__ = ABCMeta
+
+
+
+    # Superclass factory method generates the subclass instance
+    @classmethod
+    def get_instance(cls, sciImg, spectrograph, par, caliBrate,
+                 objtype, bkg_redux=False, find_negative=False, std_redux=False, show=False,
+                 setup=None, basename=None, manual=None):
+        """
+        Instantiate the Reduce subclass appropriate for the provided
+        spectrograph.
+
+        The class must be subclassed from Reduce.  See :class:`Reduce` for
+        the description of the valid keyword arguments.
+
+        Args:
+            sciImg (pypeit.images.scienceimage.ScienceImage):
+            spectrograph (:class:`pypeit.spectrographs.spectrograph.Spectrograph`):
+            par (pypeit.par.pyepeitpar.PypeItPar):
+            caliBrate (:class:`pypeit.calibrations.Calibrations`):
+            basename (str, optional):
+                Output filename used for spectral flexure QA
+            **kwargs
+                Passed to Parent init
+
+        Returns:
+            :class:`pypeit.reduce.Reduce`:
+        """
+        return next(c for c in utils.all_subclasses(FindObjects)
+                    if c.__name__ == (spectrograph.pypeline + 'FindObjects'))(
+                            sciImg, spectrograph, par, caliBrate, objtype, bkg_redux=bkg_redux,
+                            find_negative=find_negative, std_redux=std_redux, show=show,
+                            setup=setup, basename=basename,
+                            manual=manual)
+
+    def __init__(self, sciImg, spectrograph, par, caliBrate,
+                 objtype, bkg_redux=False, find_negative=False, std_redux=False, show=False,
+                 setup=None, basename=None, manual=None):
+
+        # Setup the parameters sets for this object. NOTE: This uses objtype, not frametype!
+
+        # Instantiation attributes for this object
+        self.sciImg = sciImg
+        self.spectrograph = spectrograph
+        self.objtype = objtype
+        self.par = par
+        self.caliBrate = caliBrate
+        self.scaleimg = np.array([1.0], dtype=np.float)  # np.array([1]) applies no scale
```

Comment:
@rcooke-ast can you comment?

Reply:
It was because the scale gets updated multiple times, and was easier than putting a series of `if scaleimg is not None` statements. That said, it seems like this image is being calculated, but I don't think it's being used for anything. Leave it for now, and I'll fix this in my `KCWI_fluxcal` branch.